plugin/vimteractive.vim

```diff
@@ -12,96 +12,60 @@
 " Name of the vimteractive terminal buffer
 let g:vimteractive_buffer_name = "vimteractive_buffer"
 let g:vimteractive_terminal = ''
+let g:current_term_type = ''
 
 " Variables for running the various sessions
-let g:vimteractive_ipython_command = 'ipython --matplotlib --no-autoindent'
-let g:vimteractive_python_command = 'python'
-let g:vimteractive_bash_command = 'bash'
-let g:vimteractive_zsh_command = 'zsh'
-let g:vimteractive_julia_command = 'julia'
-let g:vimteractive_maple_command = 'maple -c "interface(errorcursor=false);"'
-
-" User commands
-" =============
-
-" Commands to start a session
-command!  Iipython :call Vimteractive_session(g:vimteractive_ipython_command) 
-command!  Ipython  :call Vimteractive_session(g:vimteractive_python_command)  
-command!  Ibash    :call Vimteractive_session(g:vimteractive_bash_command)
-command!  Izsh     :call Vimteractive_session(g:vimteractive_zsh_command)
-command!  Ijulia   :call Vimteractive_session(g:vimteractive_julia_command)
-command!  Imaple   :call Vimteractive_session(g:vimteractive_maple_command)
+"
+if !has_key(g:, 'vimteractive_commands')
+	let g:vimteractive_commands = { }
+endif
+
+let g:vimteractive_commands.ipython = 'ipython --matplotlib --no-autoindent'
+let g:vimteractive_commands.python = 'python'
+let g:vimteractive_commands.bash = 'bash'
+let g:vimteractive_commands.zsh = 'zsh'
+let g:vimteractive_commands.julia = 'julia'
+let g:vimteractive_commands.maple = 'maple -c "interface(errorcursor=false);"'
+let g:vimteractive_commands.clojure = 'clojure'
+
+" Override default shells for different filetypes
+if !has_key(g:, 'vimteractive_default_shells')
+	let g:vimteractive_default_shells = { }
+endif
+
+" If 0, disable bracketed paste escape sequences
+let g:vimteractive_bracketed_paste = {
+	\ 'clojure': 0
+	\ }
+
+if !has_key(g:, 'vimteractive_loaded')
+	let g:vimteractive_loaded = 1
+
+	for term_type in keys(g:vimteractive_commands)
```

Comment:
This is really neat. Could you add a small comment indicating that you're building the individual commands -- It's probably not as obvious to a first-time reader of the plugin

Reply:
Done.